### PR TITLE
Feature/melnychenko#230checkbox address

### PR DESCRIPTION
**Pull request by myroslav-melnychenko**
checkbox created with a trick (needed research on check icon rendering issue)

### CODE REVIEW

**Comment by dmitryy90** [comment]
wrong target branch

File: src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts [hunk 1]
```diff
@@ -31,8 +37,26 @@ export class CreateContactsFormComponent implements OnInit {
     });
   }
 
+
+  duplicateForm(ob:MatCheckboxChange) {
+    this.cd.detectChanges(); //TODO: research change detector for MatCheckbox
+    if(ob.checked === true) {
```
Comment (dmitryy90) [inline src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts:43]: you can use short expression avoiding comparison boolean with boolean

**Review by dmitryy90 — CHANGES_REQUESTED** [review]

**Review by dmitryy90 — APPROVED** [review]

File: src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts [hunk 2]
```diff
@@ -31,8 +37,26 @@ export class CreateContactsFormComponent implements OnInit {
...
+        this.ActualAddressFormGroup.controls[key].setValue(undefined)
+      });
+    }
+  }
+
+
   ngOnInit(): void {
     this.passActualAddressFormGroup.emit(this.ActualAddressFormGroup);
     this.passLegalAddressFormGroup.emit(this.LegalAddressFormGroup);
+    /*this.isSameAddressControl.valueChanges.subscribe((val)=>{
```
[the remark this answers is not in the document]
Comment (litvinets) [reply src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts]: I believe this code should be updated. You can create a new FormControl (for tracking the checkbox is checked) and subscribe to these changes. The forEach method you can simplify and replace with use such methods as patchValue and resetValue from Reactive Form Groups.  
image

image

File: src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.scss [hunk 3]
```diff
@@ -1,2 +1,13 @@
 @import "src/app/shared/styles/create-form.scss";
-@import "src/app/shared/styles/validation-form.scss";
\ No newline at end of file
+@import "src/app/shared/styles/validation-form.scss";
+
+.checkbox {
```
Comment (litvinets) [inline src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.scss:4]: Please, get rid of these styles. We already have styles in the shared scss files. You need to add class " class="step-check-box". If you want to change it, please do it in the shared one, so we will have the same for all forms. 
image

**Review by litvinets — CHANGES_REQUESTED** [review]

**Pull request closed without merging**